tiatoolbox/visualization/tileserver.py

```diff
@@ -47,7 +51,39 @@ def __init__(self, title: str, layers: Dict[str, WSIReader]) -> None:
             static_folder=data._local_sample_path(Path("visualization") / "static"),
         )
         self.tia_title = title
-        self.tia_layers = layers
+
+        # generic layer names if none provided
+        if isinstance(layers, list):
+            layers = {f"layer-{i}": p for i, p in enumerate(layers)}
+        # set up the layer dict
+        layer_def = {}
+        meta = None
+        for i, key in enumerate(layers.keys()):
+            layer = layers[key]
+
+            if isinstance(layer, (str, Path)):
+                p = Path(layer)
```

Comment:
p is not a good variable name. I think this should have been flagged by pre-commit. I suggest something descriptive like `layer_path` instead.